### PR TITLE
sg ci logs: fix ci branch, improve error handling

### PR DESCRIPTION
**Pull request by bobheadxi**
- fix `getCIBranch` always failing in CI (https://buildkite.com/sourcegraph/sourcegraph/builds/115202#4d7735c5-9892-46c5-a8b1-2c514abe5cfb/65-66)
- better message for missing bk timestamps
- continue pushing logs on error
- set additional branch label from `BUILDKITE_BRANCH` only

### CODE REVIEW

File: sg.config.yaml [hunk 1]
```diff
@@ -602,7 +602,7 @@ commands:
       mkdir -p "$(dirname ${GRAFANA_LOG_FILE})"
 
       docker inspect $CONTAINER >/dev/null 2>&1 && docker rm -f $CONTAINER
-      ./docker-images/grafana/build.sh
+      IMAGE=sourcegraph/grafana:dev ./docker-images/grafana/build.sh
```
Comment (jhchabran) [inline sg.config.yaml:605]: Good catch!

File: sg.config.yaml [hunk 2]
```diff
@@ -687,6 +687,7 @@ commands:
 
   loki:
     cmd: |
+      echo "Loki: serving on http://localhost:3100"
```
Comment (jhchabran) [inline sg.config.yaml:690]: 🙏

File: dev/sg/sg_ci.go [hunk 3]
```diff
@@ -309,23 +307,60 @@ From there, you can start exploring logs with the Grafana explore panel.
...
-							return fmt.Errorf("failed to generate stream from logs for build %d job %q: %w",
-								log.JobMeta.Build, log.JobMeta.Job, err)
+							pushErrs = append(pushErrs, fmt.Sprintf("build %d job %s: %s",
+								log.JobMeta.Build, job, err))
+							continue
+						}
+
+						// Set buildkite branch if available
+						if ciBranch := os.Getenv("BUILDKITE_BRANCH"); ciBranch != "" {
+							stream.Stream.Branch = ciBranch
```
Comment (jhchabran) [inline dev/sg/sg_ci.go:335]: Non-blocking: Shouldn't we set that, just for consistency even when we're not on Buildkite?

**Review by jhchabran — APPROVED** [review]
Thank you for fixing this 🙏 !

File: dev/sg/sg_ci.go (hunk 3 above)
Comment (bobheadxi) [reply dev/sg/sg_ci.go:335]: There are too many places for the branch to come from here 😬 I think if we want to make the branch properly universal, we likely want to do another API call to get the env config of the build

File: sg.config.yaml (hunk 1 above)
Comment (bobheadxi) [reply sg.config.yaml:605]: @daxmc99 actually got this in #26969 , so I'm going to undo this haha

File: dev/sg/internal/loki/loki.go [hunk 4]
```diff
@@ -47,6 +47,10 @@ type StreamLabels struct {
 
 	App       string `json:"app"`
 	Component string `json:"component"`
+
+	// Additional metadata for CI when pushing
+
+	Branch string `json:"branch"`
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply dev/sg/internal/loki/loki.go:53]: Let's get this stably uploading on `main` first haha, but yes long-term I'd like to at least get `main-dry-run` uploading, followed by potentially all runs

**Pull request merged**